edx_name_affirmation/tasks.py

```diff
@@ -37,7 +38,7 @@ def idv_update_verified_name_task(self, attempt_id, user_id, name_affirmation_st
                 'status': name_affirmation_status
              }
              )
-    verified_names = VerifiedName.objects.filter(user__id=user_id, verified_name=photo_id_name).order_by('-created')
+    verified_names = VerifiedName.objects.filter((Q(verification_attempt_id = attempt_id) | Q(verification_attempt_id__isnull = True)) & Q(user__id = user_id) & Q(verified_name = photo_id_name)).order_by('-created')
```

Comment:
We don't want to update all verified name objects that match the user id and verified name, only the ones that either a) are not associated with any IDV attempt yet or b) are associated with the IDV attempt we got an update for.